### ISSUE TITLE
DJ001 not raised when field has type annotation

### ISSUE THREAD

**Issue by janlarres** [astral-sh/ruff#20906]
### Summary

```python
from django.db import models


class TestModel(models.Model):
    charfield1 = models.CharField(null=True) # Raises DJ001
    charfield2: models.CharField[str, str] = models.CharField(null=True) # No error
```

```
$ ruff check --select=DJ001 test.py
DJ001 Avoid using `null=True` on string-based fields such as `CharField`
 --> test.py:5:18
  |
4 | class TestModel(models.Model):
5 |     charfield1 = models.CharField(null=True)
  |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
6 |     charfield2: models.CharField[str, str] = models.CharField(null=True)
  |

Found 1 error.
```

I discovered this when using https://github.com/typeddjango/django-stubs with pyright, which doesn't detect the types correctly without explicit annotations.

### Version

ruff 0.14.0

**Comment by ntBre** [comment]
Ah yep, looks like we skip over annotated assignments:

https://github.com/astral-sh/ruff/blob/73520e4acd9cb259f41a60b0b6396c66ea32950a/crates/ruff_linter/src/rules/flake8_django/rules/nullable_model_string_field.rs#L63-L65
